src/client.ts

```diff
@@ -6408,11 +6408,11 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      * @return {Promise<string[]>} Resolves to a set of rooms
      * @return {module:http-api.MatrixError} Rejects: with an error response.
      */
-    public async _unstable_getSharedRooms(userId: string): Promise<string[]> { // eslint-disable-line
+    public async _unstable_getMutualRooms(userId: string): Promise<string[]> { // eslint-disable-line
         if (!(await this.doesServerSupportUnstableFeature("uk.half-shot.msc2666"))) {
```

Comment:
This check is no longer valid, because the flag won't have changed to support the new endpoint: old servers will be advertising that they support the API, but the code won't believe them.

A new flag is required, and both will need to be checked. It looks like the API is equivalent otherwise, so just a matter of using the now-old flag for the `shared_rooms` version and the new flag (preferring it) for the `mutual_rooms` version.